submissions/andrewklmn/task_js_dom/app.js

```diff
@@ -0,0 +1,209 @@
+};
+
+const menuItemClickListener = (event)=>{
+  let elem = event.target;
+  state.activeIndex = elem.parentNode.id;
+  renderPage(state);
+
+  const vw = Math.max(document.documentElement.clientWidth || 0, window.innerWidth || 0);
+  if (vw < 820) {
+    document.querySelector('.btn-toogle-menu').click(); 
+  }
+
+  event.stopPropagation();
+  return false;
+};
+
+const renderMenu = (state)=>{
+  
+  let nav = document.querySelector('nav');
+  let prevUl = document.querySelector('ul');
+  let fragment = document.createDocumentFragment();
+  let ul = document.createElement("ul");
+ 
+ state.items.forEach((item,index)=>{
+    let li = document.createElement("li");   
+    li = document.createElement("li");
+
+    let a = document.createElement("a");
+    li.id = index;
+      
+    a = document.createElement("a");
+    a.innerHTML = item.menu;
+    a.className = "menu-item";
+    a.href = "javascript:void(0)";
+    
+    if (index==state.activeIndex) {
+      a.classList.toggle("active");
+    };
+    
+    a.addEventListener('click',menuItemClickListener);
+    li.appendChild(a);
+    ul.appendChild(li);
+  });
+
+  fragment.appendChild(ul);
+  prevUl.remove();
+  nav.appendChild(fragment);
+};
+
+const showTextContent = ()=>{
+  document.body.style.color = "black";
+}
+
+const renderArticle = (content)=> {
+
+  document.body.style.color = "white";
+  
+  let article = document.querySelector('article');
+  article.className = "wraper";
+  let fragment = document.createDocumentFragment();
+  
+  let a = document.createElement("a");
+  a.innerHTML = "&laquo;";
+  a.className = "btn-toogle-menu";
+  a.href = "javascript:void(0)";
+  a.addEventListener('click', menuToogleClickListener);
+  fragment.appendChild(a);
+
+  let h = document.createElement("h2");
+  h.innerHTML = content.header;
+  h.className = "header-line";
+  fragment.appendChild(h);
+
+  let img = document.createElement("img");
+  img.src = content.image;
+  img.className = "picture";
+  img.alt = content.image + ' image';
+  img.addEventListener('load',showTextContent);
+  fragment.appendChild(img);
+
+  let p = document.createElement("p");
+  p.innerHTML = content.description;
+  p.className = "description"
+  fragment.appendChild(p);
+  
+  let span = document.createElement("span");
+  span.innerHTML = content.habitat;
+  span.className = "habitat"
+  p = document.createElement("p");
+  p.innerHTML = 'Habitat: ';
+
+  p.appendChild(span);
+  p.className = "habitat-line"
+  fragment.appendChild(p);
+
+  p = document.createElement("p");
+  p.innerHTML = content.method;
+  p.className = "method"
+  fragment.appendChild(p);
+  
+
+  article.innerHTML="";
+  article.appendChild(fragment);
+
+};
+
+const renderPage = (state) => {
+
+  renderMenu(state);
+  renderArticle(state.items[state.activeIndex]);
+
+};
+
+// start App when DOM is loaded
+document.addEventListener('DOMContentLoaded', (event)=>{
+
+  // Add first menu item content from default HTML layout
+  state.items.unshift({
```

Comment:
Why did you add this object to the array?
You can initially write everything to an array and display information by specifying the desired index. And not publish this information like as static in HTML

Reply:
I did it because Googlebot needs something to read for its search database. It can read it as a static HTML page without JS instead of an empty page. But I can move this content to the initial state object.
Should I move it?

Reply:
You may be right, but there are sites that are completely generated on the server
And they use other techniques for the Googlebot